client/client.go

```diff
@@ -272,6 +278,22 @@ func (c *httpClusterClient) Sync(ctx context.Context) error {
 	return c.reset(eps)
 }
 
+func (c *httpClusterClient) AutoSync(ctx context.Context, interval time.Duration) error {
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+	for {
+		err := c.Sync(ctx)
+		if err != nil {
+			return err
+		}
+		select {
+		case <-ctx.Done():
+			return nil
```

Comment:
This is the way that I want users to use to stop AutoSync elegantly, so I returns nil here. On the other hand, users could check the error with context error to see whether this matches their expectation.

I think it should return ctx.Err. I find this example: http://godoc.org/golang.org/x/net/context#Context The library wants us to return the error.